src/platformio.ts

```diff
@@ -100,6 +120,37 @@ export class PlatformIO {
         this._pioTerminal.onDidCloseTerminal(closedTerminal);
     }
 
+    private configureAutoUpdateIncludes(): void {
+        const autoUpdateIncludes = vscode.workspace.getConfiguration("platformio").get<boolean>("autoUpdateIncludes");
+        if (autoUpdateIncludes && !this._autoUpdateIncludesEnabled) {
+            const fileSystemWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**");
+            this._autoUpdateIncludesDisposables.push(fileSystemWatcher);
+            fileSystemWatcher.onDidCreate((e: vscode.Uri) => this.fileSystemDidChange(e), this, this._autoUpdateIncludesDisposables);
+            fileSystemWatcher.onDidDelete((e: vscode.Uri) => this.fileSystemDidChange(e), this, this._autoUpdateIncludesDisposables);
+            const iniFileWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**/platformio.ini");
+            this._autoUpdateIncludesDisposables.push(iniFileWatcher);
+            iniFileWatcher.onDidChange((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+            iniFileWatcher.onDidCreate((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+        }

```

Comment:
I have now changed to code, so it will watch the ```.piolibdeps``` and ```lib``` folders for created or deleted folders. Because of the challenges I had with the ```FileSystemWatcher```, I have to keep a list of existing folders to detect deletes. I wrote a rant about it here: https://github.com/Microsoft/vscode/issues/26852